router/relay/relay.go

```diff
@@ -1242,15 +1242,15 @@ func (rst *RelayStateImpl) PrepareRelayStepOnAnyRoute(cmngr poolmgr.PoolMgr) (fu
 			return rst.UnrouteRoutes(rst.CurrentRoutes())
 		}, nil
 	case ErrSkipQuery:
-		if err := rst.Client().ReplyErrMsg(err.Error()); err != nil {
+		if err := rst.Client().ReplyErr(err); err != nil {
 			return noopCloseRouteFunc, err
 		}
 		return noopCloseRouteFunc, ErrSkipQuery
 	case qrouter.MatchShardError:
-		_ = rst.Client().ReplyErrMsg("failed to match any datashard")
+		_ = rst.Client().ReplyErrMsg("failed to match any datashard", "SQPRD")
```

Comment:
What do you think if we don't duplicate error messages where possible? 

`_ = rst.Client().ReplyErrMsg(qrouter.MatchShardError, "SQPRD")`

Reply:
you dont need to specify both qrouter.MatchShardError and "SQPRD", you can specify either one